.github/workflows/sdk.yml

```diff
@@ -0,0 +1,343 @@
+          test -d sdk/core/dist && test -d sdk/msp-client/dist
+      - name: Install e2e deps (sdk/e2e)
+        working-directory: sdk/e2e
+        run: pnpm install --frozen-lockfile
+      - name: Install Playwright Chromium with system deps (sdk/e2e)
+        working-directory: sdk/e2e
+        run: |
+          pnpm exec playwright install --with-deps chromium
+      - name: Run Playwright tests (headless via xvfb) (sdk/e2e)
+        working-directory: sdk/e2e
+        run: |
+          xvfb-run -a pnpm exec playwright test --project metamask
+      - name: Upload Playwright report
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: playwright-report-wallet-metamask
+          path: /tmp/playwright-report/
+          retention-days: 14
+
+  e2e-backend-msp:
+    needs: [sdk-build-and-test]
+    name: E2E Backend Tests (MSP Client)
+    runs-on: blacksmith-16vcpu-ubuntu-2404
+    env:
+      HEADLESS: 'false'
+      RUST_LOG: info
+      SCCACHE_GHA_ENABLED: 'true'
+      RUSTC_WRAPPER: 'sccache'
+      CARGO_INCREMENTAL: '0'
+      CARGO_TERM_COLOR: always
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup pnpm & cache
+        uses: ./.github/workflow-templates/setup-pnpm
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Setup Protobuf (protoc)
+        uses: arduino/setup-protoc@v3
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup Rust toolchain (for backend build)
+        uses: actions-rust-lang/setup-rust-toolchain@v1.8
+        with:
+          cache: true
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.9
+      - name: Cache cargo registry and git index
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+      - uses: rui314/setup-mold@v1
+      - name: Set build flags (use mold linker)
+        run: echo "RUSTFLAGS=${{ env.RUSTFLAGS }} -C linker=clang -C link-arg=-fuse-ld=mold" >> $GITHUB_ENV
+      - name: Install system deps (libpq-dev, libclang-dev)
+        run: sudo apt-get update && sudo apt-get install -y libpq-dev libclang-dev
+      - name: Install dependencies (sdk workspace, no scripts)
+        working-directory: sdk
+        run: pnpm install --frozen-lockfile
+      - name: Download SDK dist
+        uses: actions/download-artifact@v4
+        with:
+          name: sdk-dist
+          path: sdk
+      - name: Place SDK dist
+        run: |
+          ls -la sdk/core || true
+          ls -la sdk/msp-client || true
+          test -d sdk/core/dist && test -d sdk/msp-client/dist
+      - name: Build backend (mocks)
```

Comment:
Not for this PR but since we have a better way to build the backend docker image, probably we can use it to run these tests.